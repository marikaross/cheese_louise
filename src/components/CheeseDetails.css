.cheese-details {
  width: 45%;
  height: 100%;
  background-image: url('../images/low-contrast-linen.png');
  color: #fefaea;
  margin: 5px;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.6);
  margin: 0 auto;
  justify-content: space-around;
  width: 75%;
  margin-top: 20px;
}

.overlay-cheese {
  background-color: rgba(54, 74, 117, .9);
  width: 100%;
  height: 100%;display: flex;
  padding-bottom: 20px;
}

.cheese-detail-image {
  width: 215px;
  height: 215px;
  -webkit-filter: grayscale(.4);
  filter: grayscale(.4);
  margin: 5px;
}

.detail-name {
  font-size: 1.5em;
}

.details {
  font-size: 1em;
}

.back {
  background-color: #fefaea;
  text-decoration: none;
  border: none;
  color: #ba3e30;
  font-size: 1.1em;
  margin: 5px 5px, 15px, 5px;
  border-radius: 7%;
  box-shadow: 0px 2px 5px 0px rgba(255, 254, 254, 0.8);
}

.cheese-deets {
  width: 60%;

}