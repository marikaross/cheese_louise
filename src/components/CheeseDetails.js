import React from 'react';
import { Link } from 'react-router-dom';
import { addFaveCheese, deleteCheese } from '../actions';
import PropTypes from 'prop-types';
import './CheeseDetails.css';
import cheeseIcon from '../images/milk.svg'


export const CheeseDetails = ({cheeseId, name, milk, region, summary, picture}) => {

  return (
      <div className='cheese-details'>
      {picture.length ? 
        <img src={`https://public.opendatasoft.com/explore/dataset/frenchcheese/files/${picture}/300/`}/> : 
        <img clasName='cheese-icon' src={cheeseIcon} />
      }
        <div className='cheese-deets'>
          <h2>{name}</h2>
          <h4>{milk}</h4>
          <h4>{region}</h4>
          <h4>{summary}</h4>
          <Link to='/'>back</Link>
        </div>
      </div>
    )
}


CheeseDetails.propTypes = {
  name: PropTypes.string,
  milk: PropTypes.string,
  region: PropTypes.string,
  summary: PropTypes.string,
  picture: PropTypes.string
}

