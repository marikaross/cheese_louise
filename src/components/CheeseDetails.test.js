import { CheeseDetails } from './CheeseDetails';
import React from 'react';
import { shallow } from 'enzyme';

describe('CheeseDetails', () => {

  it('should match the snapShot', () => {
    let wrapper = shallow(<CheeseDetails />)
    expect(wrapper).toMatchSnapshot()
  })
}) 