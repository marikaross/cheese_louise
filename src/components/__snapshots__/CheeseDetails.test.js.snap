// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheeseDetails should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    picture="hi"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="overlay-cheese"
>
        <div
                className="cheese-pic-div"
        >
                <img
                        alt="cheese"
                        className="cheese-detail-image"
                        src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
                />
        </div>
        <div
                className="cheese-deets"
        >
                <h2
                        className="detail-name"
                />
                <h4
                        className="details"
                />
                <h4
                        className="details"
                >
                        region 
                </h4>
                <h4 />
                <Link
                        className="back"
                        replace={false}
                        to="/"
                >
                        back
                </Link>
        </div>
</div>,
      "className": "cheese-details",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cheese-pic-div"
>
            <img
                        alt="cheese"
                        className="cheese-detail-image"
                        src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
            />
</div>,
          <div
            className="cheese-deets"
>
            <h2
                        className="detail-name"
            />
            <h4
                        className="details"
            />
            <h4
                        className="details"
            >
                        region 
            </h4>
            <h4 />
            <Link
                        className="back"
                        replace={false}
                        to="/"
            >
                        back
            </Link>
</div>,
        ],
        "className": "overlay-cheese",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="cheese"
              className="cheese-detail-image"
              src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
/>,
            "className": "cheese-pic-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "cheese",
              "className": "cheese-detail-image",
              "src": "https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="detail-name"
/>,
              <h4
                className="details"
/>,
              <h4
                className="details"
>
                region 
</h4>,
              <h4 />,
              <Link
                className="back"
                replace={false}
                to="/"
>
                back
</Link>,
            ],
            "className": "cheese-deets",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "detail-name",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "details",
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "region ",
                  undefined,
                ],
                "className": "details",
              },
              "ref": null,
              "rendered": Array [
                "region ",
                undefined,
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "back",
                "className": "back",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "back",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="overlay-cheese"
>
          <div
                    className="cheese-pic-div"
          >
                    <img
                              alt="cheese"
                              className="cheese-detail-image"
                              src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
                    />
          </div>
          <div
                    className="cheese-deets"
          >
                    <h2
                              className="detail-name"
                    />
                    <h4
                              className="details"
                    />
                    <h4
                              className="details"
                    >
                              region 
                    </h4>
                    <h4 />
                    <Link
                              className="back"
                              replace={false}
                              to="/"
                    >
                              back
                    </Link>
          </div>
</div>,
        "className": "cheese-details",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cheese-pic-div"
>
              <img
                            alt="cheese"
                            className="cheese-detail-image"
                            src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
              />
</div>,
            <div
              className="cheese-deets"
>
              <h2
                            className="detail-name"
              />
              <h4
                            className="details"
              />
              <h4
                            className="details"
              >
                            region 
              </h4>
              <h4 />
              <Link
                            className="back"
                            replace={false}
                            to="/"
              >
                            back
              </Link>
</div>,
          ],
          "className": "overlay-cheese",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="cheese"
                className="cheese-detail-image"
                src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/"
/>,
              "className": "cheese-pic-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "cheese",
                "className": "cheese-detail-image",
                "src": "https://public.opendatasoft.com/explore/dataset/frenchcheese/files/hi/300/",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="detail-name"
/>,
                <h4
                  className="details"
/>,
                <h4
                  className="details"
>
                  region 
</h4>,
                <h4 />,
                <Link
                  className="back"
                  replace={false}
                  to="/"
>
                  back
</Link>,
              ],
              "className": "cheese-deets",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "detail-name",
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "details",
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "region ",
                    undefined,
                  ],
                  "className": "details",
                },
                "ref": null,
                "rendered": Array [
                  "region ",
                  undefined,
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "back",
                  "className": "back",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "back",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
